### PR TITLE
feat: migrate set buy now

### PR DESCRIPTION
**Pull request by domw30**
[Associated Notion Card](https://www.notion.so/zerotech/4985d8ec0c2f4eb4860678eec483e6f6?v=7fd94d39937343aa983ac94b7d7034d5&p=45205ad2de6942edbac7803df1d61551&pm=s)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
- feature

## 3. What is the old behaviour?
Set buy now feature flow had not been migrated to zapp-nfts.

## 4. What is the new behaviour?
Set buy now migrated to zapp-nfts.

### CODE REVIEW

File: src/features/ui/ToggleButton/ToggleButton.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import styles from './ToggleButton.module.scss';
+
+type ToggleButtonProps = {
+	label?: string | React.ReactNode;
+	toggled: boolean;
+	hideOnOffLabels?: boolean;
+	onClick: () => void;
+};
+
+export const ToggleButton: FC<ToggleButtonProps> = ({
```
Comment (domw30) [inline src/features/ui/ToggleButton/ToggleButton.tsx:12]: We'll need to add a toggle button to zUI. For now I pulled this over from zns-dapp and made some minor changes.

File: src/features/view-nft/HistoryItem/HistoryItem.tsx [hunk 2]
```diff
@@ -71,7 +71,8 @@ export const HistoryItem = ({ item, tokenName }: HistoryItemProps) => {
 						secondaryAddress={item.seller}
 						amount={formatEthers(item.amount!) + tokenString}
 					/>
-					<Date timestamp={item.timestamp!} />
+					{/* NOTE: multiplying by 1000 here as transfer timestamps are in seconds for some reason */}
+					<Date timestamp={(Number(item.timestamp!) * 1000).toString()} />
```
Comment (domw30) [inline src/features/view-nft/HistoryItem/HistoryItem.tsx:75]: Random bug I spotted. Without the `*1000` it was displaying that events took place 10's of years ago

**Comment by domw30** [comment]
Made a couple more additions - added Set Buy Now option to the more options dropdown on the NFT page (if owner).

Conflicts will be resolved after Buy Now is merged to main.

https://user-images.githubusercontent.com/39112648/210845209-3e93b05c-4709-4407-b169-b3c42a61eb35.mp4

File: src/features/ui/ToggleButton/ToggleButton.tsx (hunk 1 above)
Comment (domw30) [reply src/features/ui/ToggleButton/ToggleButton.tsx:12]: I've improved this component in Migrate Domain Settings task.

**Review by padala — APPROVED** [review]
Looks good to me other than the minor things we discussed in buy now PR. Thanks

**Pull request merged**